This patch is about to increase the beacon buffer (ATH_BCBUF) and
the hw-stuck related counter (change probably not required).
Additionally, it enforces that dtim_counter gets decreased every
time - even if a beacon got stuck - because some STAs (i.e. rt2x00)
do not properly resync their sleep when beacons get stuck.
Further, beacons shall get queued within on beacon intervall regardless
of the other BSS beacon actions.

--- a/drivers/net/wireless/ath/ath9k/init.c
+++ b/drivers/net/wireless/ath/ath9k/init.c
@@ -839,7 +839,7 @@ static void ath9k_init_txpower_limits(st
 
 static const struct ieee80211_iface_limit if_limits[] = {
 	{ .max = 2048,	.types = BIT(NL80211_IFTYPE_STATION) },
-	{ .max = 8,	.types =
+	{ .max = 32,	.types =
 #ifdef CPTCFG_MAC80211_MESH
 				 BIT(NL80211_IFTYPE_MESH_POINT) |
 #endif
--- a/drivers/net/wireless/ath/ath9k/ath9k.h
+++ b/drivers/net/wireless/ath/ath9k/ath9k.h
@@ -689,7 +689,7 @@ void ath9k_set_txpower(struct ath_softc
  * number of beacon intervals, the game's up.
  */
 #define BSTUCK_THRESH           	9
-#define	ATH_BCBUF               	8
+#define	ATH_BCBUF               	32
 #define ATH_DEFAULT_BINTVAL     	100 /* TU */
 #define ATH_DEFAULT_BMISS_LIMIT 	10
 
